### DISCUSSION TITLE
Ugs doesnt conect with Arduino mega 2560 running Grbl v1.1

### DISCUSSION THREAD

**Discussion by jduribe30** [winder/Universal-G-Code-Sender#2179] [Help]
I have uploaded grbl 1.1 to a genuine Arduino mega2560 controller succesfully but when I try to cennect it to UGS, it  doesnt work. the icon in the top bar change the color like if is coneccted but it isnt, and teh setup wizards shows says that doesnt know how to configure the controller. has anyone had this issue?

![grbl uploaded](https://user-images.githubusercontent.com/35866291/226515948-14ff07a2-4d97-459f-9b74-185eea86d787.png)
![ugs](https://user-images.githubusercontent.com/35866291/226515983-d5da6133-b5f0-446d-873a-716e6861aa1a.png)
![setup wizard](https://user-images.githubusercontent.com/35866291/226516005-9d1941ee-4fb8-4ab2-98fd-c6337ae1a9ab.png)

**Comment by breiler** [comment]
I noticed similar behaviour with Grbl-Mega-5x. I believe that this has to do with a slightly longer boot time of the Mega. We are issuing a reset (CTRL-X) immediatly after connect to the controller which I think somehow disrupts the boot sequence. 

I have tried removing the reset on connect which seemed to fix the problem for the Grbl-Mega-5x. But did not want commit this change as I don't know how this will affect all other GRBL variants out there.

**Reply by jduribe30** [reply]
Happy to try your solution, how do you do that ?

**Reply by breiler** [reply]
I just retried my experiments from a couple of months ago and there was apperently a bit more to it, and I can't get it to work reliably at all now and need much more work...

Just to make sure, are you using this version of GRBL: https://github.com/gnea/grbl-Mega

**Reply by jduribe30** [reply]
Yes, that is the one

**Reply by breiler** [reply]
I think that I got it working in the latest nightly build:
https://github.com/winder/Universal-G-Code-Sender#downloads
